opal/corelib/array.rb

```diff
@@ -869,12 +869,12 @@ def drop(number)
 
   def dup
     %x{
-      if (
-        self.$$class === Opal.Array &&
-        self.$$class.$allocate.$$pristine &&
-        self.$copy_instance_variables.$$pristine &&
-        self.$initialize_dup.$$pristine
-      ) return self.slice(0);
+      if (typeof self.slice === 'function' && 
```

Comment:
I think it doesn't make sense to check that `.slice` is a function. This method gets called on the instance of Array / Array subclass, so it's always there.

Reply:
@iliabylich It's an edge case but what about `delete Array.slice` ? 🤔

Reply:
Ok, I've removed the check